ExitIntegrationTest.java

```diff
@@ -0,0 +1,176 @@
+    }
+    
+    @Test
+    public void testNormalFlowSeason() {
+        assertEquals(testExit.getState(), "IDLE");  //checks initial state
+        testIS.setCarDetected(true);
+        testExit.carEventDetected(testIS.getId(), true);
+        
+        assertEquals(testExit.getState(), "WAITING");
+        verify(testUI).display("Insert Ticket");
+        
+        when(mockSeasonDAO.findTicketById(testSeason.getId())).thenReturn(testSeason);
+        testSeason.recordUsage(mockRecord);
+        testExit.ticketInserted(testSeason.getId());
+        assertEquals(testExit.getState(), "PROCESSED");
+        
+        testExit.ticketTaken();
+        verify(testUI).display("Take Processed Ticket");
+        assertEquals(testExit.getState(), "TAKEN");     
+        
+        testOS.setCarDetected(true);
+        testExit.carEventDetected(testOS.getId(), true);
+        assertEquals(testExit.getState(), "EXITING");
+        
+        testIS.setCarDetected(false);
+        testExit.carEventDetected(testIS.getId(), false);
+        assertEquals(testExit.getState(), "EXITED");
+    }
+    
+    @Test
+    public void testNormalFlowAdhoc() {
+        assertEquals(testExit.getState(), "IDLE");  //checks initial state
+        testIS.setCarDetected(true);
+        testExit.carEventDetected(testIS.getId(), true);
+        
+        assertEquals(testExit.getState(), "WAITING");
+        verify(testUI).display("Insert Ticket");            //checks state transition and message display
+        
+        when(mockAdhocDAO.findTicketByBarcode("Atest")).thenReturn(testAdhoc);
+        testAdhoc.pay(1234, 1234);
+        testExit.ticketInserted(testAdhoc.getBarcode());
+        assertEquals(testExit.getState(), "PROCESSED");
+        
+        testExit.ticketTaken();
+        verify(testUI).display("Take Processed Ticket");
+        assertEquals(testExit.getState(), "TAKEN");     
+        
+        testOS.setCarDetected(true);
+        testExit.carEventDetected(testOS.getId(), true);
+        assertEquals(testExit.getState(), "EXITING");
+        
+        testIS.setCarDetected(false);
+        testExit.carEventDetected(testIS.getId(), false);
+        assertEquals(testExit.getState(), "EXITED");
+    }
+    
+    @Test
+    public void testAlternateFlowAdhoc() {          //test accurate detection of rejected adhoc ticket
+        assertEquals(testExit.getState(), "IDLE");  //checks initial state
+        
+        testIS.setCarDetected(true);
+        testExit.carEventDetected(testIS.getId(), true);
+        assertEquals(testExit.getState(), "WAITING");
+        
+        when(mockAdhocDAO.findTicketByBarcode("Atest")).thenReturn(null);
+        testAdhoc.pay(1234, 1234);
+        testExit.ticketInserted(testAdhoc.getBarcode());
+        assertEquals(testExit.getState(), "REJECTED");
+    }
+    
+    @Test
+    public void testAlternateFlowSeason() {          //test accurate detection of rejected season ticket

```

Comment:
This method name is not very descriptive of the actual test being performed

Reply:
Fixed.